### PR TITLE
Implement read timeout for PFCP connection to detect a peer crash

### PR DESCRIPTION
**Pull request by osinstom**
If a PFCP does not send any message in the `read_timeout` period, PFCP Agent closes the PFCP connection and clears the fastpath state.

This behavior is needed to detect SMF crash (covered by `smf-failure.xml` STC test).

### CODE REVIEW

File: pfcpiface/conn.go [hunk 1]
```diff
@@ -179,12 +179,26 @@ func (pConn *PFCPConn) setLocalNodeID(id string) {
...
 		for {
+			err := pConn.SetReadDeadline(time.Now().Add(pConn.upf.readTimeout))
+			if err != nil {
+				log.Errorf("failed to set read timeout: %v", err)
+			}
+
 			n, err := pConn.Read(recvBuf)
 			if err != nil {
+				if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+					log.Infof("Read timeout received for connection: %v<->%v",
```
Comment (ccascone) [inline pfcpiface/conn.go:195]: Thinking about who will have to interpret these logs:
```suggestion
					log.Infof("Read timeout for connection %v<->%v, is the SMF still alive?",
```

File: pfcpiface/conn.go [hunk 2]
```diff
@@ -179,12 +179,26 @@ func (pConn *PFCPConn) setLocalNodeID(id string) {
 
 // Serve serves forever a single PFCP peer.
 func (pConn *PFCPConn) Serve() {
-	go func() {
+	connClosed := make(chan struct{}, 1)
```
Comment (ccascone) [inline pfcpiface/conn.go:182]: The name is misleading, it looks like a read timeout can happen even with a connection that is still open
```suggestion
	connTimeout := make(chan struct{}, 1)
```

**Review by ccascone — APPROVED** [review]
Looking good. Just left a few minor comments.

**Review by ccascone — DISMISSED** [review]

**Comment by osinstom** [comment]
retest this please

**Comment by osinstom** [comment]
retest this please

**Comment by osinstom** [comment]
retest this please

**Comment by osinstom** [comment]
retest this please

**Review by ccascone — APPROVED** [review]

**Pull request merged**